datafusion/src/physical_plan/sort_preserving_merge.rs

```diff
@@ -965,7 +965,7 @@ mod tests {
                 options: Default::default(),
             },
             PhysicalSortExpr {
-                expr: col("c7", &schema).unwrap(),
+                expr: col("c12", &schema).unwrap(),
```

Comment:
The original `c7` is not distinguishable enough, i.e. same value exists many times.

Reply:
👍  The same thing happens in arrow 6.0 (as pointed out by @houqp ) -- more details here: https://github.com/apache/arrow-datafusion/pull/984/files#r705557467